### DISCUSSION TITLE
Why would I want to turn off flymake when using eglot?

### DISCUSSION THREAD

**Discussion by jgarte** [brotzeit/rustic#549] [Q&A]
Hi,

In the following section:

https://github.com/brotzeit/rustic#eglot

It mentions 

> Turn off flymake.

But it doesn't say why the user should do that. 

Is this because eglot will manage flymake directly?

I want to use flymake and not flycheck when using eglot.

Just wanted to confirm.

Would it be good to mention why the user is turning off flymake in the README when using eglot for clarity?

**Comment by mak-kirkland** [comment]
I'm wondering the same thing... anyone have an answer?

**Reply by mak-kirkland** [reply]
Digging around I think we actually want to keep flymake ENABLED when using eglot. 

eglot uses flymake to display the warnings, hints etc. that eglot fetches from the lsp server (the rust-analyzer).

if we turn off flymake, then we're turning off all the inline and hover suggestions... which entirely defeats the purpose of using eglot in for syntax checking in the first place...
